source/OutfitterPanel.cpp

```diff
@@ -330,427 +362,573 @@ double OutfitterPanel::DrawDetails(const Point &center)
+					player.Accounts().AddCredits(-price);
+					player.AddStock(selectedOutfit, -1);
+
+					// Install it on this ship.
+					ship->AddOutfit(selectedOutfit, 1);
+					int required = selectedOutfit->Get("required crew");
+					if(required && ship->Crew() + required <= static_cast<int>(ship->Attributes().Get("bunks")))
+						ship->AddCrew(required);
+					ship->Recharge();
+				}
 			}
-			else
+		}
+		else if(toLocation == OutfitLocation::Cargo)
+		{
+			// Buy up to <modifier> of the selected outfit and place them in fleet cargo.
+			double mass = selectedOutfit->Mass();
+			if(mass)
+				howManyPer = min(howManyPer, static_cast<int>(player.Cargo().FreePrecise() / mass));
+
+			// How much will it cost to buy all that we can fit?
+			int64_t price = player.StockDepreciation().Value(selectedOutfit, day, howManyPer);
+
+			// Adjust the number to buy to stay within the player's available credits.
+			while(price > player.Accounts().Credits() && howManyPer)
 			{
-				int64_t price = player.FleetDepreciation().Value(selectedOutfit, day);
-				player.Accounts().AddCredits(price);
-				player.AddStock(selectedOutfit, 1);
+				// The current cost is more than the amount of money the player has, so try a smaller amount.
+				if(!--howManyPer)
+					// If the smaller amount becomes zero, we're not able to do anything with cargo.
+					return "Cannot afford to buy this item and load into cargo.";
+				price = player.StockDepreciation().Value(selectedOutfit, day, howManyPer);
 			}
 
-			const Outfit *ammo = selectedOutfit->AmmoStoredOrUsed();
-			if(ammo && ship->OutfitCount(ammo))
+			if(howManyPer)
 			{
-				// Determine how many of this ammo I must sell to also sell the launcher.
-				int mustSell = 0;
-				for(const pair<const char *, double> &it : ship->Attributes().Attributes())
-					if(it.second < 0.)
-						mustSell = max<int>(mustSell, it.second / ammo->Get(it.first));
+				// Buy as many as can be paid for and will fit.
+				player.Accounts().AddCredits(-price);
+				player.AddStock(selectedOutfit, -howManyPer);
 
-				if(mustSell)
+				// Put them into fleet cargo.
+				player.Cargo().Add(selectedOutfit, howManyPer);
+			}
+		}
+		// Note: Buying into storage not implemented. Why waste your money?
+	}
+	else if(fromLocation == OutfitLocation::Ship)
+	{
+		for(int i = howManyPer; i; i--)
+		{
+			// Get the ships that have the most of this outfit installed.
+			const vector<Ship *> shipsToOutfit = GetShipsToOutfit();
+			// Note: to get here, we have already confirmed that every ship in the selection
+			// has the outfit installed and that the outfit can be uninstalled in the first place.
+			if(shipsToOutfit.empty())
+				break;
+			for(Ship *ship : shipsToOutfit)
+			{
+				// Uninstall the outfit.
+				ship->AddOutfit(selectedOutfit, -1);
+				int required = selectedOutfit->Get("required crew");
+				if(required)
+					ship->AddCrew(-required);
+				// Adjust hired crew counts.
+				ship->Recharge();
+
+				// If the context is sale:
```

Comment:
Not every line requires its own comment and this one is pretty self explanatory. We have descriptive variable names for the exact reason so we don't have to put a comment on every other line.
https://github.com/endless-sky/endless-sky/wiki/C++-Style-Guide#comments
```suggestion
```